src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -177,4 +186,119 @@ public ExamResource(UserService userService, ExamService examService, ExamAccess
+                if (!user.get().getGroups().contains(course.getStudentGroupName())) {
+                    userService.addUserToGroup(user.get(), course.getStudentGroupName());
+                }
+                continue;
+            }
+            // 2) if we cannot find the user, we use the registration number and try to find the student in the (TUM) LDAP
+            user = userService.createUserFromLdap(registrationNumber);
+            if (user.isPresent()) {
+                exam.addUser(user.get());
+                // the newly created user needs to get the rights to access the course, otherwise the student cannot access the exam (within the course)
+                userService.addUserToGroup(user.get(), course.getStudentGroupName());
+                continue;
+            }
+            // 3) if we cannot find the user in the (TUM) LDAP, we report this to the client
+            log.warn("User with registration number " + registrationNumber + " not found in Artemis user database and not found in (TUM) LDAP");
+            notFoundStudents.add(student);
+        }
+        examRepository.save(exam);
+        return ResponseEntity.ok().body(notFoundStudents);
+    }
+
+    /**
+     * DELETE /courses/:courseId/exams/:examId/students/:studentLogin : Remove one single given user (based on the login) from the students of the exam so that the student cannot access the exam any more
+     *
+     * @param courseId     the id of the course
+     * @param examId       the id of the exam
+     * @param studentLogin the login of the user who should lose student access
+     * @return empty ResponseEntity with status 200 (OK) or with status 404 (Not Found)
+     */
+    @DeleteMapping(value = "/courses/{courseId}/exams/{examId}/students/{studentLogin:" + Constants.LOGIN_REGEX + "}")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Void> removeStudentFromExam(@PathVariable Long courseId, @PathVariable Long examId, @PathVariable String studentLogin) {
+        log.debug("REST request to remove {} as student from exam : {}", studentLogin, examId);
+        var course = courseService.findOne(courseId);
+        var instructorOrAdmin = userService.getUserWithGroupsAndAuthorities();
+        if (!authCheckService.isAtLeastInstructorInCourse(course, instructorOrAdmin)) {
+            return forbidden();
+        }
+        var exam = examService.findOneWithRegisteredUsers(examId);
+        if (!course.equals(exam.getCourse())) {
+            return forbidden();
+        }
+        Optional<User> student = userService.getUserWithGroupsAndAuthoritiesByLogin(studentLogin);
+        if (student.isEmpty()) {
+            return notFound();
+        }
+        exam.removeUser(student.get());
+        examRepository.save(exam);
+        return ResponseEntity.ok().body(null);
```

Comment:
Why should this be a problem? I guess this is fine, because then the request worked. This call should be used when the instructor clicks on Remove in the mockups for one user. This does not automatically remove the user from the course.

 It could be the case that students are registered for a course, but not for an exam.

What we actually need on the other hand is that students who are added to the exam, also have access to the course. I will add this above.